### ISSUE TITLE
Error when updating cluster Code="InvalidIdentityValues"

### ISSUE THREAD

**Issue by faraday23** [Azure-Terraform/terraform-azurerm-kubernetes#54]
### Expected Behavior

I'm trying to plan/apply with my existing aks cluster  
I am using latest version of terraform-azurerm-kubernetes v3.2.1

### Actual Behavior

throws error (see Error output)

### Error Output

```
Error: updating Managed Kubernetes Cluster "aks-app-ueplakgkpkxc-sandbox-eastus2"  (Resource Group "app-ueplakgkpkxc-sandbox-eastus2"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=0 
-- Original Error: Code="InvalidIdentityValues" Message="Invalid value for the identities '/subscriptions/b0837458-adf3-41b0-a8fb-c16f9719627d/resourceGroups/app-ueplakgkpkxc-sandbox-eastus2/providers/Microsoft.ManagedIdentity/userAssignedIdentities/uai-aks-app-ueplakgkpkxc-sandbox-eastus2'. 
The 'UserAssignedIdentities' property keys should only be empty json objects, null or the resource existing property."

  on .terraform\modules\kubernetes\main.tf line 31, in resource "azurerm_kubernetes_cluster" "aks":
  31: resource "azurerm_kubernetes_cluster" "aks" {
```

### I check the state file and the ID field looks correct

```hcl
      "module": "module.kubernetes",
      "mode": "managed",
      "type": "azurerm_user_assigned_identity",
      "name": "aks",
      "provider": "provider[\"registry.terraform.io/hashicorp/azurerm\"]",
      "instances": [
        {
          "index_key": 0,
          "schema_version": 1,
          "attributes": {
            "client_id": "24d0f8e4-476d-4ec7-a8c6-66806c76b862",
            "id": "/subscriptions/b0837458-adf3-41b0-a8fb-c16f9719627d/resourceGroups/app-ueplakgkpkxc-sandbox-eastus2/providers/Microsoft.ManagedIdentity/userAssignedIdentities/uai-aks-app-ueplakgkpkxc-sandbox-eastus2",
            "location": "eastus2",
            "name": "uai-aks-app-ueplakgkpkxc-sandbox-eastus2",
            "principal_id": "c0d56213-f9ae-4124-a2b2-daf1a204bb06",
```

### Affected Resource(s)

* `azurerm_v2.40.0`
* `terraform v0.13.5`

### Terraform Configuration Files

**Main.tf**  
```hcl
# Modules for Kubernetes Cluster #
module "kubernetes" {
  source = "github.com/Azure-Terraform/terraform-azurerm-kubernetes.git?ref=v3.2.1"

  location            = module.metadata.location
  names               = module.metadata.names
  tags                = module.metadata.tags
  resource_group_name = module.resource_group.name

  identity_type = "UserAssigned"

  windows_profile = {
    admin_username = "azadmin"
    admin_password = random_password.admin.result
  }

  kubernetes_version         = "1.19.7"
  network_plugin             = "kubenet"
  dns_prefix                 = "rest-server"
  configure_network_role     = true
  configure_subnet_nsg_rules = true
  enable_kube_dashboard      = false

  #acr_pull_access = {
  #  "sre-registry" = "/subscriptions/32496c5b-1147-452c-8469-3a11028f8946/resourceGroups/sre_container_registry/providers/Microsoft.ContainerRegistry/registries/srecontainerregistry"
  #}

  default_node_pool = "green"

  node_pools = {
    system = {
      subnet     = "private"
      vm_size    = "Standard_B2s"
      node_count = 2
    }
    blue = {
      vm_size             = "Standard_B2ms"
      enable_auto_scaling = true
      min_count           = 1
      max_count           = 3
      subnet              = "public"
    }
    green = {
      vm_size             = "Standard_B2ms"
      enable_auto_scaling = true
      min_count           = 1
      max_count           = 3
      subnet              = "public"
    }
  }

  node_pool_subnets = {
    private = {
      id                          = module.virtual_network.subnets["iaas-private"].id
      resource_group_name         = module.virtual_network.subnets["iaas-private"].resource_group_name
      network_security_group_name = module.virtual_network.subnets["iaas-private"].network_security_group_name
    }
    public = {
      id                          = module.virtual_network.subnets["iaas-public"].id
      resource_group_name         = module.virtual_network.subnets["iaas-public"].resource_group_name
      network_security_group_name = module.virtual_network.subnets["iaas-public"].network_security_group_name
    }
  }
}
```

**Comment by fabiendelpierre** [comment]
It's a known issue between Hashicorp's azurerm provider and the Azure API -- not a fault with this module. 

https://discuss.hashicorp.com/t/user-assigned-identities-property-keys-should-only-be-empty-json-objects-null-or-the-resource-exisiting-property/20962

https://github.com/Azure/azure-rest-api-specs/issues/13631

https://github.com/terraform-providers/terraform-provider-azurerm/issues/10406

Basically you're likely to hit this issue if you try to modify an existing AKS cluster.

Microsoft and Hashicorp are arguing over the way to fix it.

**Comment by faraday23** [comment]
Just to possibly help others experiencing the same issue..

I was able to reproduce this issue when I made changes to my Kubernetes cluster. So it has absolutely nothing to do with "InvalidIdentityValues". (It is a false alarm error)

If another person runs into this issue, whatever change you made to your cluster, if you revert the changes and run Terraform plan/apply it will work. (not a solution but a workaround)

**Comment by jamurtag** [comment]
Looks like there is a workaround as documented [here](https://github.com/terraform-providers/terraform-provider-azurerm/issues/10406#issuecomment-822641196).

@dutsmiller - can this be implemented temporarily?

**Comment by dutsmiller** [comment]
@jamurtag I implemented it and tested it.  If we merge that code TF will try to make a change on EVERY apply going forward even if the code hasn't changed.  Is this what we really want to do?

**Comment by dutsmiller** [comment]
btw here is the "hack":

`  identity {
    type                      = var.identity_type
    user_assigned_identity_id = (var.identity_type == "SystemAssigned" ? null :
                                (var.user_assigned_identity != null ? 
                                 replace(var.user_assigned_identity.id,"resourceGroups","resourcegroups") : 
                                 replace(azurerm_user_assigned_identity.aks.0.id,"resourceGroups","resourcegroups")))
  }`

**Comment by dutsmiller** [comment]
https://github.com/Azure-Terraform/terraform-azurerm-kubernetes/releases/tag/v3.2.2
